src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -114,11 +125,25 @@ private void Initialize(Compilation comp, AnalyzerExecutor analyzerExecutor, Can
             }
         }
 
-        private Task ExecuteSyntaxTreeActions(CancellationToken cancellationToken)
+        public void StartCompleteAnalysis(ImmutableHashSet<SyntaxTree> syntaxTreesToSkip, CancellationToken cancellationToken)
+        {
+            if (_initializeTaskStarted && _primaryTask == null)
+            {
+                _primaryTask = Task.Run(async () =>
+                    {
+                        await _initializeTask.ConfigureAwait(false);
+                        await ProcessCompilationEventsAsync(true, cancellationToken).ConfigureAwait(false);
+                        await ExecuteSyntaxTreeActions(_compilation.SyntaxTrees.Where((tree) => !syntaxTreesToSkip.Contains(tree)), cancellationToken).ConfigureAwait(false);
```

Comment:
I recently moved the ExecuteSyntaxTreeActions invocation outside the primary task as it was being executed after compilation end action. Its now inside ProcessCompilationEventsAsync but awaited before the end actions are executed. Probably when you merge, you should delete this line.